homeassistant/components/knocki/coordinator.py

```diff
@@ -19,16 +21,37 @@ def __init__(self, hass: HomeAssistant, client: KnockiClient) -> None:
             name=DOMAIN,
         )
         self.client = client
+        self._known_triggers: set[tuple[str, int]] = set()
 
     async def _async_update_data(self) -> dict[int, Trigger]:
         try:
             triggers = await self.client.get_triggers()
         except KnockiConnectionError as exc:
             raise UpdateFailed from exc
+        current_triggers = {
+            (trigger.device_id, trigger.details.trigger_id) for trigger in triggers
+        }
+        removed_triggers = self._known_triggers - current_triggers
+        for trigger in removed_triggers:
+            await self._delete_device(trigger)
+        self._known_triggers = current_triggers
         return {trigger.details.trigger_id: trigger for trigger in triggers}
 
     def add_trigger(self, event: Event) -> None:
         """Add a trigger to the coordinator."""
         self.async_set_updated_data(
             {**self.data, event.payload.details.trigger_id: event.payload}
         )
+        self._known_triggers.add(
+            (event.payload.device_id, event.payload.details.trigger_id)
+        )
+
+    async def _delete_device(self, trigger: tuple[str, int]) -> None:
```

Comment:
This doesn't need to be a coroutine function since there's no await inside. Make it an async callback.